website/addons/owncloud/static/owncloudNodeConfig.js

```diff
@@ -0,0 +1,315 @@
+                return false; // Prevent event propagation
+            },
+            connectAccount: function() {
+                window.location.href = this.urls().auth;
+            },
+            decodeFolder: function(folder_name) {
+                return folder_name;
+            }
+        };
+        // Overrides
+        self.options = $.extend({}, defaults, opts);
+        // Treebeard config
+        self.treebeardOptions = $.extend(
+            {},
+            OauthAddonFolderPicker.prototype.treebeardOptions,
+            {
+                onPickFolder: function(evt, item) {
+                    return this.options.onPickFolder.call(this, evt, item);
+                }.bind(this),
+                resolveLazyloadUrl: function(item) {
+                    return item.data.urls.folders;
+                },
+                decodeFolder: function(item) {
+                    return this.options.decodeFolder.call(this, item);
+                }.bind(this)
+
+            }
+        );
+
+        self.credentialsChanged = ko.pureComputed(function() {
+            return self.nodeHasAuth() && !self.validCredentials();
+        });
+        self.showCredentialInput = ko.pureComputed(function() {
+            return (self.credentialsChanged() && self.userIsOwner()) ||
+                (!self.userHasAuth() && !self.nodeHasAuth() && self.loadedSettings());
+        });
+
+        self.showSettings = ko.pureComputed(function() {
+            return self.nodeHasAuth() && self.validCredentials();
+        });
+        self.showImport = ko.pureComputed(function() {
+            return self.userHasAuth() && !self.nodeHasAuth() && self.loadedSettings();
+        });
+        self.showCredentialCreateButton = ko.pureComputed(function() {
+            return !self.userHasAuth() && !self.nodeHasAuth() && self.loadedSettings();
+        });
+
+        // Flashed messages
+        self.message = ko.observable('');
+        self.messageClass = ko.observable('text-info');
+
+        // Update above observables with data from the server
+        $.ajax({
+            url: url,
+            type: 'GET',
+            dataType: 'json'
+        }).done(function(response) {
+            // Update view model
+            if(response.result.folder){
+                self.folderName(response.result.folder.name);
+            }
+            self.updateFromData(response.result);
+            self.loadedSettings(true);
+        }).fail(function(xhr, textStatus, error) {
+            self.changeMessage(language.userSettingsError, 'text-danger');
+            Raven.captureMessage('Could not GET ownCloud settings', {
+                url: url,
+                textStatus: textStatus,
+                error: error
+            });
+        });
+
+        self.selectionChanged = function() {
+            self.changeMessage('','');
+        };
+    }
+});
+
+
+/** Reset all fields from OwnCloud host selection modal */
+ViewModel.prototype.clearModal = function() {
```

Comment:
It seems like you're doing more work in this class than you need to. Take a look at the updates to `s3NodeConfig.js`; it's doing almost exactly what you need, just with a different modal that accepts an API key rather than username and password.